src/openrct2-ui/windows/CustomCurrency.cpp

```diff
@@ -42,204 +42,182 @@ static rct_widget window_custom_currency_widgets[] = {
-{
-    auto ft = Formatter::Common();
-    ft.Add<money64>(MONEY(10, 0));
+        DrawWidgets(dpi);
 
-    WindowDrawWidgets(w, dpi);
+        auto screenCoords = windowPos + ScreenCoordsXY{ 10, 30 };
 
-    auto screenCoords = w->windowPos + ScreenCoordsXY{ 10, 30 };
+        DrawTextBasic(&dpi, screenCoords, STR_RATE, {}, { colours[1] });
 
-    DrawTextBasic(dpi, screenCoords, STR_RATE, {}, { w->colours[1] });
+        int32_t baseExchange = CurrencyDescriptors[EnumValue(CurrencyType::Pounds)].rate;
+        ft = Formatter();
+        ft.Add<int32_t>(baseExchange);
+        DrawTextBasic(&dpi, screenCoords + ScreenCoordsXY{ 200, 0 }, STR_CUSTOM_CURRENCY_EQUIVALENCY, ft, { colours[1] });
 
-    int32_t baseExchange = CurrencyDescriptors[EnumValue(CurrencyType::Pounds)].rate;
-    ft = Formatter();
-    ft.Add<int32_t>(baseExchange);
-    DrawTextBasic(dpi, screenCoords + ScreenCoordsXY{ 200, 0 }, STR_CUSTOM_CURRENCY_EQUIVALENCY, ft, { w->colours[1] });
+        screenCoords.y += 20;
 
-    screenCoords.y += 20;
+        DrawTextBasic(&dpi, screenCoords, STR_CURRENCY_SYMBOL_TEXT, {}, { colours[1] });
 
-    DrawTextBasic(dpi, screenCoords, STR_CURRENCY_SYMBOL_TEXT, {}, { w->colours[1] });
+        screenCoords = windowPos
+            + ScreenCoordsXY{ window_custom_currency_widgets[WIDX_SYMBOL_TEXT].left + 1,
+                              window_custom_currency_widgets[WIDX_SYMBOL_TEXT].top };
 
-    screenCoords = w->windowPos
-        + ScreenCoordsXY{ window_custom_currency_widgets[WIDX_SYMBOL_TEXT].left + 1,
-                          window_custom_currency_widgets[WIDX_SYMBOL_TEXT].top };
+        gfx_draw_string(
+            &dpi, screenCoords, CurrencyDescriptors[EnumValue(CurrencyType::Custom)].symbol_unicode, { colours[1] });
 
-    gfx_draw_string(dpi, screenCoords, CurrencyDescriptors[EnumValue(CurrencyType::Custom)].symbol_unicode, { w->colours[1] });
+        auto drawPos = windowPos
+            + ScreenCoordsXY{ window_custom_currency_widgets[WIDX_AFFIX_DROPDOWN].left + 1,
+                              window_custom_currency_widgets[WIDX_AFFIX_DROPDOWN].top };
+        rct_string_id stringId = (CurrencyDescriptors[EnumValue(CurrencyType::Custom)].affix_unicode == CurrencyAffix::Prefix)
+            ? STR_PREFIX
+            : STR_SUFFIX;
+        DrawTextBasic(&dpi, drawPos, stringId, {}, { colours[1] });
+    }
+};
 
-    auto drawPos = w->windowPos
-        + ScreenCoordsXY{ window_custom_currency_widgets[WIDX_AFFIX_DROPDOWN].left + 1,
-                          window_custom_currency_widgets[WIDX_AFFIX_DROPDOWN].top };
-    rct_string_id stringId = (CurrencyDescriptors[EnumValue(CurrencyType::Custom)].affix_unicode == CurrencyAffix::Prefix)
-        ? STR_PREFIX
-        : STR_SUFFIX;
-    DrawTextBasic(dpi, drawPos, stringId, {}, { w->colours[1] });
+rct_window* CustomCurrencyWindowOpen()
+{
+    return WindowFocusOrCreate<CustomCurrencyWindow>(WC_CUSTOM_CURRENCY_CONFIG, 400, 100, 0);
```

Comment:
use WW and WH macros for the width and height.
Also this is not creating the window centered. You need to add the flag `WF_CENTRE_SCREEN`

Reply:
That was missing in the original code and I wondered about it, thanks!